agent/api/api_client.go

```diff
@@ -77,8 +78,20 @@ func (client *ApiECSClient) serviceClient() (*svc.AmazonEC2ContainerServiceV2014
 	return ecs, nil
 }
 
+// serviceClient provides a client for interacting with the ECS service apis
+func (client *ApiECSClient) serviceClient() (svc.AmazonEC2ContainerServiceV20141113, error) {
+	return client.serviceClientFn()
+}
+
 func NewECSClient(credentialProvider credentials.AWSCredentialProvider, config *config.Config, insecureSkipVerify bool) ECSClient {
```

Comment:
Minor: I'd probably reorder this so NewECSClient is first, serviceClient is second, and serviceClientImpl is third.